### ISSUE TITLE
Remove the FooService.NewClient factory method from generated code.

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#6986]
From GA API review feedback:
There's a constructor that takes channel as an argument, so the method seems extraneous. It's unclear if the method is actually useful, let's remove it before GA (can be reintroduced later as a non-breaking change).

**Comment by jtattermusch** [comment]
I started making the change and I'm a bit unhappy with the result. 

Instead of simply
`Math.NewClient(channel)`  I ended up with  `new Math.MathClient(channel)` and `new TestService.TestServiceClient(channel)`  (and I expect the "Service" suffix to be pretty common for service names).

For C# 6 that has `using static` this is fine because once can write just `new TestServiceClient()`, but for older C# versions the code ends up being a bit ugly. I think there should be a concise way to create a new client and `NewClient` serves the purpose.

So, after rethinking this issue, it seems that by this change we would be increasing code (and documentation) churn and removing an existing API (breaking some users, although trivially) only to provide a less convenient way to express the same concept. It doesn't seem anyone's going to benefit from this.

@jskeet  @chrisdunelm @mattdr  - I suggest leaving the factory method as is. Opinions?
